calendar-bundle/src/Resources/contao/modules/ModuleCalendar.php

```diff
@@ -119,7 +119,8 @@ protected function compile()
 		$time = Date::floorToMinute();
 
 		// Find the boundaries
-		$objMinMax = $this->Database->query("SELECT MIN(startTime) AS dateFrom, MAX(endTime) AS dateTo, MAX(repeatEnd) AS repeatUntil FROM tl_calendar_events WHERE pid IN(" . implode(',', array_map('\intval', $this->cal_calendar)) . ")" . (!BE_USER_LOGGED_IN ? " AND published='1' AND (start='' OR start<='$time') AND (stop='' OR stop>'$time')" : ""));
+		$blnShowUnpublished = System::getContainer()->get('contao.security.token_checker')->isPreviewMode();
```

Comment:
https://github.com/aschempp/contao/blob/fca680e747bc0e924d1e1a6130f49932cc7102c8/core-bundle/src/Security/Authentication/Token/TokenChecker.php#L113

`TokenChecker::isPreviewMode()` only checks for an available `FrontendUser`,  but the framework defines `BE_USER_LOGGED_IN` based on the backend user:

https://github.com/aschempp/contao/blob/fca680e747bc0e924d1e1a6130f49932cc7102c8/core-bundle/src/Framework/ContaoFramework.php#L188

Do I miss something here?

Reply:
The `FrontendPreviewToken` is used for backend users (currently). See https://github.com/contao/contao/blob/7daf6b8b67ee86d7c1a89d404cdb78fd247427da/core-bundle/src/Controller/BackendPreviewSwitchController.php#L128

Reply:
My issue is that IMO you cannot replace 

```
\define('BE_USER_LOGGED_IN', $this->tokenChecker->hasBackendUser() && $this->tokenChecker->isPreviewMode());
```
by 
```
tokenChecker->isPreviewMode()
```

Reply:
The name `BE_USER_LOGGED_IN` is confusing, since it only ever represented the frontend preview mode (show unpublished items). If a backend user was logged in but unpublished items were not enabled, the constant would actually resolve to `false`.

also related to https://github.com/contao/contao/pull/3872

Reply:
Thanks for clarifying! 👍